checks.d/spark.py

```diff
@@ -0,0 +1,381 @@
+        running_apps = self._get_running_spark_apps(rm_address)
+
+        # Report success after gathering all metrics from ResourceManaager
+        self.service_check(YARN_SERVICE_CHECK,
+            AgentCheck.OK,
+            tags=['url:%s' % rm_address],
+            message='Connection to ResourceManager "%s" was successful' % rm_address)
+
+        # Get the job metrics
+        self._spark_job_metrics(running_apps, cluster_id)
+
+        # Get the stage metrics
+        self._spark_stage_metrics(running_apps, cluster_id)
+
+        # Get the executor metrics
+        self._spark_executor_metrics(running_apps, cluster_id)
+
+        # Get the rdd metrics
+        self._spark_rdd_metrics(running_apps, cluster_id)
+
+        # Report success after gathering all metrics from the ApplicationMaster
+        if running_apps:
+            app_id, (app_name, tracking_url) = running_apps.items()[0]
+            am_address = self._get_url_base(tracking_url)
+
+            self.service_check(SPARK_SERVICE_CHECK,
+                AgentCheck.OK,
+                tags=['url:%s' % am_address],
+                message='Connection to ApplicationMaster "%s" was successful' % rm_address)
+
+    def _get_cluster_id(self, rm_address):
+        '''
+        Return the cluster ID, otherwise raise an exception
+        '''
+        info_json = self._rest_request_to_json(rm_address,
+            INFO_PATH,
+            YARN_SERVICE_CHECK)
+
+        return info_json.get('clusterInfo', {}).get('id')
+
+    def _get_running_spark_apps(self, rm_address):
+        '''
+        Return a dictionary of {app_id: (app_name, tracking_url)} for the running Spark applications
+        '''
+        metrics_json = self._rest_request_to_json(rm_address,
+            YARN_APPS_PATH,
+            YARN_SERVICE_CHECK,
+            states=YARN_APPLICATION_STATES,
+            applicationTypes=YARN_APPLICATION_TYPES)
+
+        running_apps = {}
+
+        if metrics_json.get('apps'):
+            if metrics_json['apps'].get('app') is not None:
+
+                for app_json in metrics_json['apps']['app']:
+                    app_id = app_json.get('id')
+                    tracking_url = app_json.get('trackingUrl')
+                    app_name = app_json.get('name')
+
+                    if app_id and tracking_url and app_name:
+                        running_apps[app_id] = (app_name, tracking_url)
+
+        return running_apps
+
+    def _spark_job_metrics(self, running_apps, cluster_id):
+        '''
+        Get metrics for each Spark job.
+        Return a map from Stage IDs to Job IDs
+        '''
+        for app_id, (app_name, tracking_url) in running_apps.iteritems():
+
+            response = self._rest_request_to_json(tracking_url,
+                SPARK_JOBS_PATH,
+                SPARK_SERVICE_CHECK, app_name, 'jobs')
+
+            tags = ['cluster_id:' + str(cluster_id),
+                    'app_name:' + str(app_name)]
+
+            if len(response) > 0:
```

Comment:
This condition is unnecessary (even without it, no iteration of the `for` loop below will be run if `len(response) == 0`).

Applies to all the other `if len(response)` in the check too ;)